client/utils/RangedParticleEmitter.mjs

```diff
@@ -0,0 +1,113 @@
+import Phaser from "phaser";
+import { getObjectTiles } from "../../shared/utils/index.mjs";
+
+export class RangedParticleEmitter {
+  constructor({
+    scene,
+    particleImgId,
+    positionTile,
+    mainObjectSize,
+    timerDelay = 50,
+    timerStartAt = 50,
+    timerLoop = true,
+    maxParticlesPerc = 50,
+    particleDuration = 5000,
+    sizeToIncrease = { x: 1, y: 1 },
+    withoutMainObject = false,
+  }) {
+    this.scene = scene;
+    this.particleImgId = particleImgId;
+    this.particleDuration = particleDuration;
+
+    this.rangeTiles = this.scene.groundLayer.getTilesWithin(
+      positionTile.tileX - 1 - sizeToIncrease.x,
+      positionTile.tileY - 1 - sizeToIncrease.y,
+      sizeToIncrease.x * 2 + mainObjectSize.x,
+      sizeToIncrease.y * 2 + mainObjectSize.y
+    );
+
+    if (withoutMainObject) {
```

Comment:
Something wrong with the naming, when no main object then get object titles using it's size? Isn't it enough to use `mainObjectSize` instead?